internal/internal_task_handlers.go

```diff
@@ -830,6 +852,19 @@ process_Workflow_Loop:
 	return response, err
 }
 
+// ProcessWorkflowTask processes the given workflow which includes
+// - fetching, reordering and replaying historical decision events. (Decision events in this context is an umbrella term for workflow relevant events)
+// - state machine is incrementally built with every decision.
+// - state machine makes sure that when a workflow restarts for some reason same activities (or timers etc.) are not called again and previous result state is loaded into memory
+//
+// Note about Replay tests mode:
```

Comment:
@Groxx / @agautam478 This PR is not fully ready to be reviewed but requesting early feedback from you two. 
I feel like the false positives are mostly handled with this fix and `make test` is passing. Not sure how much I should trust on the test coverage yet :) We can also have a quick call tomorrow to discuss if that works.

Reply:
the tests are not reliable indicators of the changes especially in Replay tests. Yes, let's get on a call.

Reply:
scheduled for tomorrow